source/_posts/durable-entities.md

```diff
@@ -0,0 +1,129 @@
+
+- Http Client Function(Stateless) - Receives the request, validates it, Sets up the Orchestrator.
+- Orchestrator - Has the flow logic and triggers Activities.
+- Activities - This has the actual compute logic and does the work.
+
+For more information on these topics please see [here](https://docs.microsoft.com/nl-nl/azure/azure-functions).
+
+## What is an Entity?
+
+An Entity is what u expect it to be. If you worked with Entity-Framework or DDD before it should be a familiar term. An entity is a class consisting of properties that represents a set of data in some kind of storage.
+
+In our case, the state of the Entity is persisted in Azure Table Storage by the framework.  Entities in Durable Entities are a bit different in the sense that they are like small microservices (Actors) that have methods on them that you can call. In order to call an Entity we need to know its `Entity ID`. This Id uniquely identifies the Entity. An Entity-Id consists of a name and a key that together form a uniquely identifiable string.
+
+There are also other properties on an Entity like:
+
+**Operation name**, this would be the name of an operation that can be performed. For example `AssignBedAsync`.
+
+**Operation Input**, These are parameters that can be provided with the operation to perform. These parameters are optional. For example the name of the person that will occupy the bed.
+
+**Scheduled time**, This optional parameter indicates the specific time of delivery. Operations can be scheduled to run in the future using this parameter.
+
+A very simple Entity looks like this:
+
+```csharp
+public class BedEntity
+{
+    public string BedNumber { get; set; }
+    public bool IsOccupied { get; set; }
+    
+    private readonly ILogger _logger;
+
+    public BedEntity(ILogger logger, string bedNumber)
+    {
+        _logger = logger;
+        BedNumber = bedNumber;
+    }
+    
+    [FunctionName(nameof(BedEntity))]
+    public static async Task HandleEntityOperation(
+        [EntityTrigger] IDurableEntityContext context,
+        ILogger logger)
+    {
+        await context.DispatchAsync<BedEntity>(logger, context.EntityKey);
+    }
+    
+    public Task<bool> IsOccupiedBedAsync()
+    {
+        var IsOccupied = BedNumber == "123";
+        return Task.FromResult(IsOccupied);
+    }
+    
+    public Task AssignBedAsync(string bedNumber)
+    {
+        IsOccupied = true;
+        return Task.CompletedTask;
+    }
+}
+```
+
+If we look at the above code you might be thinking but how and where does the call to the database take place to store the `BedEntity` after its modified? This is all abstracted by the framework. We do not need to worry about it. Both methods `IsOccupiedBedAsync` and `AssignBedAsync` change the state of the entity asynchronously and then the framework does the rest. The function `HandleEntityOperation` allows us to call operations on this entity from the client function or orchestrator.
```

Comment:
...asynchronously and then the framework does the rest.

asynchronously while leaving the state management and database actions to the framework.


niet helemaal zeker maar **does the rest** is spreektaal(?)